packages/core-flows/src/definition/cart/workflows/add-to-cart.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  AddToCartWorkflowInputDTO,
+  CreateLineItemForCartDTO,
+} from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import {
+  addToCartStep,
+  getVariantPriceSetsStep,
+  getVariantsStep,
+  validateVariantsExistStep,
+} from "../steps"
+import { prepareLineItemData } from "../utils/prepare-line-item-data"
+
+// TODO: The AddToCartWorkflow are missing the following steps:
+// - Confirm inventory exists (inventory module)
+// - Refresh/delete shipping methods (fulfillment module)
+// - Create line item adjustments (promotion module)
+// - Update payment sessions (payment module)
+
+export const addToCartWorkflowId = "add-to-cart"
+export const addToCartWorkflow = createWorkflow(
+  addToCartWorkflowId,
+  (input: WorkflowData<AddToCartWorkflowInputDTO>) => {
+    const variantIds = transform({ input }, (data) => {
+      return (data.input.items ?? []).map((i) => i.variant_id)
+    })
+
+    validateVariantsExistStep({ variantIds })
+
+    // TODO: This is on par with the context used in v1.*, but we can be more flexible.
+    const pricingContext = transform({ cart: input.cart }, (data) => {
+      return {
+        currency_code: data.cart.currency_code,
+        region_id: data.cart.region_id,
+        customer_id: data.cart.customer_id,
```

Comment:
Not important for now, but this might need customer_group_id as well to activate price_list scoped queries.